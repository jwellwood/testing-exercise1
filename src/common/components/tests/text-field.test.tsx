import * as React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { TextField } from '../forms/text-field';

describe('Form components test', () => {
  let inputElem;
  beforeEach(() => {
    const inputProps = {} as never;
    const metaProps = {};
    render(<TextField input={inputProps} meta={metaProps} />);
    inputElem = screen.getByRole('textbox') as HTMLInputElement;
  });
  test('should match snapshot', () => {
    expect(inputElem).toMatchSnapshot();
  });
  test('should render a text field', () => {
    expect(inputElem).toBeInTheDocument();
  });
  test('should have empty value without user interaction', () => {
    expect(inputElem).toHaveValue('');
  });
  test('should update value of text field when user enters text', () => {
    const userEnteredValue = 'Hello World!';
    fireEvent.change(inputElem, { target: { value: userEnteredValue } });
    expect(inputElem).toHaveValue(userEnteredValue);
  });
  test('should not show error paragraph if metadata does not have error', () => {
    const inputProps = {} as never;
    const metaProps = {};
    const { container } = render(
      <TextField input={inputProps} meta={metaProps} />
    );
    expect(container.querySelector('p')).not.toBeInTheDocument();
  });
  test('should show error paragraph if metadata has error', () => {
    const inputProps = {} as never;
    const metaProps = {
      error: 'error',
      touched: true,
      submitError: true,
      dirtySinceLastSubmit: false,
    };
    render(<TextField input={inputProps} meta={metaProps} />);
    expect(screen.getByText('error')).toBeInTheDocument();
  });
});
