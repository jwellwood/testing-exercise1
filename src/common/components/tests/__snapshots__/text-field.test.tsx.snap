// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form components test should match snapshot 1`] = `
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiInput-input"
  type="text"
  value=""
/>
`;
